src/js/ttsHookup.js

```diff
@@ -38,51 +38,73 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
-                    }
-                }
+    fluid.defaults("gpii.firstDiscovery.tts.prefsEditor", {
+        gradeNames: ["gpii.firstDiscovery.msgLookup"],
+        invokers: {
+            speakPanelMessage: {
+                funcName: "gpii.firstDiscovery.tts.prefsEditor.speakPanelMessage",
+                args: ["{firstDiscoveryEditor}", "{that}.msgLookup.stepCountMsg", "{that}.msgLookup.panelMsg", "{gpii.firstDiscovery.selfVoicing}.queueSpeech", "{arguments}.0"]
+            },
+            speakPanelInstructions: {
+                funcName: "gpii.firstDiscovery.tts.prefsEditor.speakPanelInstructions",
+                args: ["{firstDiscoveryEditor}", "{gpii.firstDiscovery.selfVoicing}.queueSpeech", "{arguments}.0"]
+            }
+        },
+        listeners: {
+            "onCreate.bindKeypress": {
+                listener: "gpii.firstDiscovery.keyboardShortcut.bindShortcut",
+                args: ["body", gpii.firstDiscovery.keyboardShortcut.key.h, [], "{that}.speakPanelInstructions"]
+            },
+            "onReady.speakPanelMessage": {
+                listener: "{that}.speakPanelMessage",
+                priority: "last"
+            }
+        },
+        modelRelay: [{
+            source: "preferences.gpii_firstDiscovery_speak",
+            target: "{gpii.firstDiscovery.selfVoicing}.model.enabled",
+            backward: "never",
+            singleTransform: {
+                type: "fluid.transforms.identity"
+            }
+        }, {
+            source: "preferences.gpii_firstDiscovery_language",
+            target: "{gpii.firstDiscovery.selfVoicing}.model.utteranceOpts.lang",
+            backward: "never",
+            singleTransform: {
+                type: "fluid.transforms.identity"
+            }
+        }, {
+            source: "preferences.gpii_firstDiscovery_speechRate",
+            target: "{gpii.firstDiscovery.selfVoicing}.model.utteranceOpts.rate",
+            backward: "never",
+            singleTransform: {
+                type: "fluid.transforms.identity"
+            }
+        }],
+        modelListeners: {
+            "{firstDiscoveryEditor}.model.currentPanelNum": {
+                listener: "{that}.speakPanelMessage",
+                excludeSource: "init"
             }
         },
+        messageBase: "{messageLoader}.resources.prefsEditor.resourceText",
         panelInstructionsSelector: ".gpiic-fd-instructions"
     });
 
-    gpii.firstDiscovery.tts.fdHookup.getCurrentPanelInstructions = function (that) {
+    gpii.firstDiscovery.tts.prefsEditor.getCurrentPanelInstructions = function (that) {
         var panel = that.panels.eq(that.model.currentPanelNum - 1);
-        var texts = fluid.transform(panel.find(that.options.panelInstructionsSelector).filter(":visible"), function (elem) {
+        var texts = fluid.transform(panel.find(that.prefsEditor.options.panelInstructionsSelector).filter(":visible"), function (elem) {
```

Comment:
I believe this is because of the panels like the keyboard assessment that have multiple sets of instructions that are shown/hidden at different times.